reana/reana_dev/utils.py

```diff
@@ -578,6 +580,42 @@ def update_module_in_cluster_components(
         )
 
 
+def upgrade_requirements(component: str) -> bool:
+    """Update the Python requirements file using pip-compile."""
+    requirements_path = os.path.join(get_srcdir(component), PYTHON_REQUIREMENTS_FILE)
+    if not os.path.exists(requirements_path):
+        display_message(f"File {PYTHON_REQUIREMENTS_FILE} not found.", component)
+        return False
+
+    pip_compile_cmd = None
+    with open(requirements_path) as requirements_file:
+        # find pip-compile command contained in requirements.txt
+        for line in requirements_file:
+            stripped_line = line.lstrip("#").strip()
+            if stripped_line.startswith("pip-compile"):
+                pip_compile_cmd = stripped_line
+                break
+
+    if not pip_compile_cmd:
+        display_message(
+            f"File {PYTHON_REQUIREMENTS_FILE} does not contain a valid pip-compile command.",
+            component,
+        )
+        return False
+
+    if "annotation-style" not in pip_compile_cmd:
+        executable, options = pip_compile_cmd.split(maxsplit=1)
```

Comment:
Just a note: probably quite unlikely, but if `pip_compile_cmd` is only composed by the string `"pip-compile"`, this will raise a `ValueError` because there aren't enough values to unpack. Do you think it's an edge case worth handling?